README.md

```diff
@@ -462,3 +462,214 @@ func main() {
 }
 ```
 </details>
+
+<details>
+<summary>Generate Embeddings</summary>
```

Comment:
There are a couple of problems with these examples:

1. They are too specific, containing non-relevant parts of file reading, binary decoding, and sorting
2. I would prefer to keep README neutral, without specific mentions of developers and their preferences
3. Both examples are not that different from being included together

I suggest we include a single minimal example that:
1. Takes a bunch of strings
2. Computes their embeddings
3. Finds the one closest to a query

And let's use fine food reviews [dataset](https://www.kaggle.com/datasets/snap/amazon-fine-food-reviews) that OpenAI has in their [docs](https://platform.openai.com/docs/guides/embeddings/use-cases):

```go
examples := []string{
	"It was tasty and fresh. The other one I bought was old and tasted moldy. But this one was good.",
	"Great coffee at a good price. I'm a subscription buyer and I buy this month after month. What more can I say?",
	"This chocolate is amazing..I love the taste and smell, this is the only chocolate for me...I found a new love!"
	"I love this coffee!  And such a great price.  Will buy more when I am running out which will be soon.",
	"The Raspberry Tea Syrup is great. I can use it for hot and cold drinks as well in certain recipes.",
	"Everyone that dips with this loves it!  So easy to use! Olive oil and tasty bread is all you need.",
	"This is a favorite of mine for using over ice. Even bought it to give out as Christmas gifts last year.",
	"If you like a great , hot, sauce then buy this. If spicy with heat isn't to your liking then don't buy it.",
}

query := "chocolate"
```

Reply:
@sashabaranov  The purpose of providing binary storage is to simulate the vector database scenario. If it is only compared by []string, it does not conform to the normal business process.

Reply:
@aceld I think computing embeddings for all `examples []string` would be sufficient for the README example.